hardware/gpsPi/gps_reader.py

```diff
@@ -52,3 +52,28 @@ def get_geolocation(self):
             return data
         else:
             return None
+
+    def get_speed_mph(self):
+        while self.gps.inWaiting() == 0:
+            pass
+
+        nmeaSentence = str(self.gps.readline()).split(",")
+        nmeaType = nmeaSentence[0]
+
+        # verify nmeaSentence type and verify if gps has valid data
+        if nmeaType == "b'$GPRMC" and nmeaSentence[2] == "A":
+
+            speed_in_knots = nmeaSentence[7]
+            speed_in_mph = float(speed_in_knots) * 1.151
+
+            self.logging.info("speed: " + str(speed_in_mph))
+
+            data = {}
+            data["sensor_id"] = GPS_ID
+            data["values"] = {
+                "speed": speed_in_mph,
+            }
+            data["date"] = date_str_with_current_timezone()
+            return data
+        else:
+            return None
```

Comment:
I think this "return None" case should be covered so that we can know whether the valid data check can handle exceptions correctly.

BTW, the line 38 is also not covered. Please update the test case so that we can check the gps_readler get south latitude correctly.